### PR TITLE
Endpoint Composition - rootNode on Endpoint 0 (fullFamily)

### PR DESCRIPTION
**Pull request by paulr34**
**Action to Fetch and Load Root Node Data**
Description:
This PR introduces the loadComposition action to the Vuex store, which performs the following tasks:

Fetch Root Node Data: Retrieves root node data from the server.
Add Endpoint Type: Calls addEndpointType with the fetched data.
Add Endpoint: Calls addEndpoint with the data returned from addEndpointType

### CODE REVIEW

File: src-electron/db/query-feature.js [hunk 1]
```diff
@@ -31,6 +31,9 @@ const dbMapping = require('./db-mapping.js')
  * with associated device type and cluster details
  */
 async function getFeaturesByDeviceTypeRefs(db, deviceTypeRefs) {
+  if (!Array.isArray(deviceTypeRefs)) {
+    deviceTypeRefs = [deviceTypeRefs]
+  }
   let deviceTypeRefsSql = deviceTypeRefs.map(() => '?').join(', ')
```
Comment (ethanzhouyc) [inline src-electron/db/query-feature.js:37]: This change is fine for the function itself, but we might want to figure out why it is not an array only in the case of rootNode

Comment (paulr34) [reply src-electron/db/query-feature.js:37]: I agree, will figure out cause before making any decisions

Comment (paulr34) [reply src-electron/db/query-feature.js]: fixed, thank you @ethanzhouyc

File: src-electron/db/query-endpoint.js [hunk 2]
```diff
@@ -59,6 +60,27 @@ ORDER BY E1.ENDPOINT_IDENTIFIER
...
+ *
+ * @param {*} db - The database connection object.
+ * @param {Array<number>} packageIds - An array of package IDs to query.
+ * @returns {Promise<Array>} - A promise that resolves to an array of rows containing DEVICE_TYPE_REF.
+ */
+async function getRootNode(db, packageIds) {
+  const query = `
+    SELECT DEVICE_TYPE_REF, CODE, NAME
+    FROM ENDPOINT_COMPOSITION
+    WHERE TYPE = ? AND PACKAGE_REF IN (${packageIds.map(() => '?').join(', ')})
```
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/db/query-endpoint.js:75]: I read it was the best practice but am not married to the idea

File: src-electron/db/zap-schema.sql [hunk 3]
```diff
@@ -386,7 +386,10 @@ CREATE TABLE IF NOT EXISTS "ENDPOINT_COMPOSITION" (
   "DEVICE_TYPE_REF" integer,
   "TYPE" text,
   "CODE" integer,
+  "NAME" text,
+  "PACKAGE_REF" integer,
```
[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/db/zap-schema.sql:390]: name is needed for creating the endpoint in the front end code and yes you're right ill fix thank you

[the remark this answers is not in the document]
Comment (paulr34) [reply src-electron/db/zap-schema.sql]: fixed, thank you

File: src/App.vue [hunk 4]
```diff
@@ -283,6 +325,9 @@ export default defineComponent({
 
       // load initial UC component state
       this.$store.dispatch(`zap/loadUcComponentState`)
+      if (this.uiThemeCategory == 'matter' && query[`newConfig`]) {
+        this.loadRootNode()
```
Comment (tecimovic) [inline src/App.vue:329]: Don't "load root node". UI should always just ask backed: "What should I do?"

**Review by tecimovic — APPROVED** [review]

File: src/App.vue (hunk 4 above)
Comment (paulr34) [reply src/App.vue]: fixed, thank you

**Pull request merged**